src/track/serato/beatsimporter.cpp

```diff
@@ -24,56 +24,83 @@ bool SeratoBeatsImporter::isEmpty() const {
 
 QVector<double> SeratoBeatsImporter::importBeatsAndApplyTimingOffset(
         const QString& filePath, const audio::StreamInfo& streamInfo) {
+    const audio::SignalInfo& signalInfo = streamInfo.getSignalInfo();
+    const double timingOffsetMillis = SeratoTags::guessTimingOffsetMillis(
+            filePath, signalInfo);
+
+    return importBeatsAndApplyTimingOffset(timingOffsetMillis, signalInfo);
+}
+
+QVector<double> SeratoBeatsImporter::importBeatsAndApplyTimingOffset(
+        double timingOffsetMillis, const audio::SignalInfo& signalInfo) {
```

Comment:
Since timingOffsetMillis depends on signalInfo I would prefer to pass it as the second parameter.

Reply:
On the other hand it replaces the file parameter. Doesn't really matter, at your choice.

Reply:
> On the other hand it replaces the file parameter. Doesn't really matter, at your choice.

Yeah, that was my reasoning. Doesn't really matter IMHO. The only reason this is a separate method is that we don't have a file path in when testing.